LanguageFeatures/Extension-types/upper_bound_A01_t06.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The Dart 1 algorithm which is used to compute the standard upper
+/// bound of two distinct interface types will work in the same way as it does
+/// today, albeit on a slightly different superinterface graph:
+///
+/// For the purpose of this algorithm, we consider `Object?` to be a
+/// superinterface of `Null` and `Object`, and the path lengths mentioned in the
+/// algorithm will increase by one (because they start from `Object?` rather
+/// than from `Object`).
+///
+/// @description Check that the least upper bound calculation algorithm
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import '../../Utils/static_type_helper.dart';
+
+extension type ET1<T>(T id) {}
+
+extension type ET2<T extends String>(T id) implements ET1<T>, String {}
+extension type ET3<T extends String>(T id) implements ET1<T?>, String {}
+
+main() {
+  var v1 = 2 > 1 ? ET2("ET2") : ET3("ET3");
+  v1.expectStaticType<Exactly<String?>>();
```

Comment:
Can't be `String?` -- shared superinterfaces are `String`, `Comparable<String>`, `Object`, `Object?`, and `String` wins.